File: backend/Domain/Entities/Post.cs
using backend.Domain.Common;

namespace backend.Domain.Entities;

public class Post : BaseEntity
{
    public string Content { get; set; } = String.Empty;
    public string? ImageUrl { get; set; }
    public int NrOfReports { get; set; }
    public bool IsPrivate { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime DateUpdated { get; set; }
    public int UserId { get; set; }

    public User User { get; set; } = new User();
    public ICollection<Like> Likes { get; set; } = new List<Like>();
    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    public ICollection<Favorite> Favorites { get; set; } = new List<Favorite>();
    public ICollection<Report> Reports { get; set; } = new List<Report>();
}

File: frontend/src/lib/types/post.ts
export interface User {
	id: number
	name: string
	email?: string
	avatar?: string
}

export interface Post {
	id: number
	content: string
	imageUrl?: string
	nrOfReports: number
	isPrivate: boolean
	isDeleted: boolean
	dateCreated: string
	dateUpdated: string
	userId: number
	user: User
	likes: Like[]
	comments: Comment[]
	favorites: Favorite[]
}

export interface Like {
	id: number
	userId: number
	postId: number
	dateCreated: string
}

export interface Comment {
	id: number
	content: string
	userId: number
	postId: number
	user: User
	dateCreated: string
	dateUpdated: string
}

export interface Favorite {
	id: number
	userId: number
	postId: number
	dateCreated: string
}

export interface CreatePostDTO {
	content: string
	imageUrl?: string
	isPrivate?: boolean
}
